aten/src/ATen/core/jit_type.h

```diff
@@ -651,10 +653,15 @@ struct CAFFE2_API TensorType : public Type {
     return strides;
   }
 
+  static std::tuple<std::vector<bool>, std::vector<size_t>>
+  contiguityStrideIndices(at::IntArrayRef sizes, at::IntArrayRef strides);
+
   c10::optional<at::ScalarType> scalar_type_;
   c10::optional<at::Device> device_;
-  VaryingShape sizes_;
-  VaryingStrides strides_;
+  VaryingShape<ShapeSymbol> sizes_;
+  VaryingShape<int64_t> strides_;
```

Comment:
* stride and contiguity information should probably go in its own object not flattened out in the tensor. 
* stride_indices_ ,and contiguity_, and strides_ need documentation. I think you can take a lot of the documentation from the quip document that proposes this interface.

Reply:
> stride and contiguity information should probably go in its own object not flattened out in the tensor.

Can we defer this piece until a bit later? I understand we want a `StrideDimension` object that has an index and contiguity but since there isn't any code that actually uses this information (e.g. contiguity and stride_indices) yet.

Reply:
Well, there at least should be tests that this information is getting tracked correctly. I found what seem like several bugs tracking this information. If it is unused, then it shouldn't be going into the PR.